runtime/lua/vim/shared.lua

```diff
@@ -401,24 +401,28 @@ end
 --- for example from |rpcrequest()| or |vim.fn|.
 ---
 ---@param t Table
----@returns `true` if array-like table, else `false`.
+---@return boolean @`true` if list (table indexed with sequential integers starting from 1), else `false`.
```

Comment:
better move the definition to the doc text above, and leave `list-like table` here